shared/types/shared.d.ts

```diff
@@ -68,6 +68,63 @@ export type DirectiveDefinition<T, D = DirectiveLocations> = {
 	directiveLocations: D;
 };
 
+export type ObjectLikeDefinition<Interfaces> = {
+	description?: string; // Description of the object type
+	isActivated?: boolean; // If the object type is activated
+	implementsInterfaces?: Interfaces; // Interfaces that the object type implements
+	typeDirectives?: DirectivePropertyData[]; // Directives for the type
```

Comment:
As far as I know, we don't parse SDL directives to raw directive types. I think it is better to use `StructuredDirective[]` type

Reply:
UPD: as we reuse it in FE logic, maybe better to move it to generic as `Interfaces`

Reply:
I moved it to generic parameters.